src/transformers/integrations.py

```diff
@@ -1537,6 +1541,63 @@ def on_save(self, args, state, control, **kwargs):
             self._clearml_task.update_output_model(artifact_path, iteration=state.global_step, auto_delete_file=False)
 
 
+class FlyteCallback(TrainerCallback):
+    """A [`TrainerCallback`] that sends the logs to [Flyte](https://flyte.org/).
+    NOTE: This callback only works within a Flyte task.
+
+    Args:
+        save_log_history (bool, optional, defaults to `True`):
+            Determines whether or not to save the training logs to the task's Flyte Deck.
+
+        sync_checkpoints (bool, optional, defaults to `True`):
+            When set to True, checkpoints are synced to Flyte and can be used to resume training in the case of an
+            interruption.
+
+    Example:
+        ```python
+        # Note: This example skips over some setup steps for brevity.
+        from flytekit import current_context, task
+
+
+        @task
+        def train_hf_transformer():
+            cp = current_context().checkpoint
+            trainer = Trainer(..., callbacks=[FlyteCallback()])
+            output = trainer.train(resume_from_checkpoint=cp.restore())
+        ```
+    """
+
+    def __init__(self, save_log_history: bool = True, sync_checkpoints: bool = True):
+        super().__init__()
+        if not is_flytekit_available():
+            raise ImportError("FlyteCallback requires flytekit to be installed. Run `pip install flytekit`.")
+
+        from flytekit import current_context
+
+        self.cp = current_context().checkpoint
+        self.save_log_history = save_log_history
+        self.sync_checkpoints = sync_checkpoints
+
+    def on_save(self, args, state, control, **kwargs):
+        if self.sync_checkpoints and state.is_world_process_zero:
+            ckpt_dir = f"checkpoint-{state.global_step}"
+            artifact_path = os.path.join(args.output_dir, ckpt_dir)
+
+            logger.info(f"Saving checkpoint in {ckpt_dir} to Flyte. This may take time.")
+            self.cp.save(artifact_path)
+
+    def on_train_end(self, args, state, control, **kwargs):
+        if self.save_log_history:
+            import pandas as pd
+            from flytekit import Deck
+
+            if is_pandas_available():
+                Deck("Log History", pd.DataFrame(state.log_history).to_html())
```

Comment:
This can be expensive if decks are not enabled. We ideally want to avoid this he expensive to_html call. Can we use the renderer instead

Reply:
Thanks for the feedback 😄 

1. Are you suggesting to use the [TableRenderer](https://github.com/flyteorg/flytekit/blob/master/plugins/flytekit-deck-standard/flytekitplugins/deck/renderer.py#L96)
2. Do you know if I can check if desks are disabled from inside a Flyte task?

Reply:
yes, I think we should use TableRenderer instead. rendered only generate html when deck is enabled.

Reply:
Done thanks for the context!